DataFormats/Portable/interface/PortableObject.h

```diff
@@ -0,0 +1,21 @@
+#ifndef DataFormats_Portable_interface_PortableObject_h
+#define DataFormats_Portable_interface_PortableObject_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+// This header is not used by PortableObject, but is included here to automatically
+// provide its content to users of ALPAKA_ACCELERATOR_NAMESPACE::PortableObject.
+#include "HeterogeneousCore/AlpakaInterface/interface/AssertDeviceMatchesHostCollection.h"
```

Comment:
@makortel for the `PortableCollection` case we include this file in `DataFormats/Portable/interface/alpaka/PortableCollection.h` rather than `DataFormats/Portable/interface/PortableCollection.h`.

Which one do you think is a better choice ?

Reply:
I think the `#include` should be in the one in the `alpaka` subdirectory, because the macro defined there makes use of `ALPAKA_ACCELERATOR_NAMESPACE`.

Reply:
It doesn't use it explicitly, the check is
```c++
  static_assert(std::is_same_v<alpaka_serial_sync::DEVICE_COLLECTION, HOST_COLLECTION>,
                "The device collection for the host device and the host collection must be the same type!");
```
but yes, the check kind of makes sense only when `ALPAKA_ACCELERATOR_NAMESPACE` is used to define different `DEVICE_COLLECTION`s.